### PR TITLE
Feature/refactor

### PR DESCRIPTION
**Pull request by EroAuditore**
- Refactor the Calculator component (components/Calculator.js) from a class based component to a functional component with hooks.
- Refactor the App component (components/app.js) from a class based component to a functional component.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -25,220 +22,218 @@ class Calculator extends Component {
...
   };
 
-  handleClick = (btnName) => {
-    const { next, total, operation } = this.state;
+  const handleClick = (btnName) => {
+    const { next, total, operation } = calcState;
 
     const obj = {
       total,
       next,
```
Comment (Davidosky007) [inline src/components/Calculator.js:30]: * I get this error when i perform divide by zero operations kindly fix it 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi, @EroAuditore, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/Calculator.js (hunk 1 above)
Comment (DeliceLydia) [reply src/components/Calculator.js:30]: Well resolved this! Good job..

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

## Hi @EroAuditore , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


### Keep Rocking!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**